visualiser/edge.py

```diff
@@ -0,0 +1,123 @@
+import copy
+from typing import Union
+
+from visualiser import Node
+
+
+class Edge:
+
+    def __init__(self, source_node: Union[Node, str], destination_node: Union[Node, str], label: str = '',
+                 **attrs: str) -> None:
+        self._source_node = Node(source_node) if isinstance(source_node, str) else copy.deepcopy(source_node)
+        self._destination_node = Node(destination_node) if isinstance(destination_node, str) else copy.deepcopy(
+            destination_node)
```

Comment:
**Suggestion:** Consider adding error handling for when the source_node or destination_node is not of type Node or str. This will prevent the program from crashing when an invalid type is passed.
```suggestion
    def __init__(self, source_node: Union[Node, str], destination_node: Union[Node, str], label: str = '',
                     **attrs: str) -> None:
            if not isinstance(source_node, (Node, str)) or not isinstance(destination_node, (Node, str)):
                raise TypeError('source_node and destination_node must be of type Node or str')
            self._source_node = Node(source_node) if isinstance(source_node, str) else copy.deepcopy(source_node)
            self._destination_node = Node(destination_node) if isinstance(destination_node, str) else copy.deepcopy(
                destination_node)
```